protoc-gen-grpc-gateway/main.go

```diff
@@ -65,8 +63,11 @@ func main() {
 		ParamFunc: flag.CommandLine.Set,
 	}.Run(func(plugin *protogen.Plugin) error {
 		// FIXME: still needed to parse request parameter and apply flags manually, see the comment above.
-		parseFlags(reg, plugin.Request.GetParameter())
-		if err := applyFlags(reg); err != nil {
+		pkgMap := parseFlags(plugin.Request.GetParameter())
+
+		gen, reg := gengateway.New(*useRequestContext, *registerFuncSuffix, *pathType, *allowPatchFeature, *standalone)
```

Comment:
Could we keep creating the `reg` outside of `gengateway` and just keep this new order of operations? E.g. still pass in the registry and mutate it, but do it before running `applyFlags`? It feels a bit dirty for `gengateway.New` to return the registry. Looks to me like this should be solved by just moving it above `parseFlags` and `applyFlags`. Am I missing something?

Reply:
yep, no problem. i was originally trying to enforce structurally in the code that it couldn't be placed out of order again, but now that we have the generation path exercised with ABE the same issue would be caught by compilation during testing.